kmip/tests/unit/core/primitives/test_interval.py

```diff
@@ -15,35 +15,211 @@
 
 import testtools
 
+from kmip.core import exceptions
+from kmip.core import primitives
 from kmip.core import utils
 
 
 class TestInterval(testtools.TestCase):
+    """
+    Test suite for the Interval primitive.
+    """
 
     def setUp(self):
         super(TestInterval, self).setUp()
-        self.stream = utils.BytearrayStream()
+
+        # Encoding and value taken from Section 9.1.2 of the KMIP 1.1
+        # specification.
+        self.value = 864000
+        self.encoding = (
+            b'\x42\x00\x00\x0A\x00\x00\x00\x04\x00\x0D\x2F\x00\x00\x00\x00'
```

Comment:
Good eye! You're seeing the discrepancy with the third byte I believe, \x20 (in the spec) vs \x00 (here in the test). This is actually ok.

For these example encodings, the spec actually gives everything the dummy tag (the first 3 bytes of the encoding) \x42\x00\x20 (see the leading text in Section 9.1.2 of the spec). This tag is actually a legitimate tag for a CompromiseDate object (see Table 213, specifically pg. 107 of the spec). However, I didn't want all of my default primitives to have the tag for a legitimate KMIP object; I want a neutral tag that can apply to anything. So instead, I made up a new default tag, \x42\x00\x00, which is specifically unused by the spec. By default, all PyKMIP primitives start with this default tag; it's up to their instantiations or subclasses to specify their real tags, for whatever type of KMIP objects they actually represent. As a result, all of the encodings I take from the spec for these primitive tests are tweaked to make sure their tags line up properly with what my primitives will actual encode to. Hence the discrepancy.